src/datasource.ts

```diff
@@ -227,14 +227,18 @@ export default class HeroicDatasource {
   }
 
   public targetContainsTemplate(target): boolean {
-    for (const group of target.groupBy) {
-      for (const param of group.params) {
-        if (this.templateSrv.variableExists(param)) {
-          return true;
+    // Check aggregations for variables.
+    for (const select of target.select) {
+      for (const aggregation of select) {
+        for (const param of aggregation.params) {
+          if (this.templateSrv.variableExists(param)) {
+            return true;
+          }
         }
       }
     }
 
+    // Check filters for variables.
```

Comment:
wait so this is supposed to work? re: can't filter w/ variables

Reply:
@lmuhlha this function `targetContainsTemplate` get's called from the grafana alert panel. what this function does now is verify aggregations and filters **do not** contain variables when the query is used in an alert. If they do a warning is shown.

Reply:
ah gotcha 👍

Reply:
maybe a more useful comment would be "ensure filters do not contain variables" but 🤷‍♀